src/main/java/org/jabref/gui/Globals.java

```diff
@@ -109,6 +109,11 @@ public static void shutdownThreadPools() {
         if (fileUpdateMonitor != null) {
             fileUpdateMonitor.shutdown();
         }
+
+        if (stateManager.getDirectoryMonitor() != null) {
+            stateManager.getDirectoryMonitor().shutdown();
+        }
```

Comment:
@calixtus Can you check if the file `DirectoryMonitor` should reside in `StateManager` or if we should move it to `Globals` (similar to `fileUpdateMonitor`).

Reply:
Can the File update monitor and the DirectoryMonitor be merged to a more generic class?

Reply:
Note that Globals points to `StateManager`:

```java
/**
 * @deprecated try to use {@link StateManager} and {@link org.jabref.preferences.PreferencesService}
 */
```

---

We need a "Global" DirectoryMonitor. Would you place it in `org.jabref.gui.DefaultInjector` and then add a `shutdown` method to it?

We need to have a full lifecycle of `DirectoryMonitor`. Creation, running, and shutdown. There currently is no migration guide at `org.jabref.gui.Globals#shutdownThreadPools`

To keep things going, I vote to move the currently directoryMonitor code into `Globals` and do a refactoring towards dependency injection afterwards.

---

Also the more generic ones migration away from `FileUpdateMonitor` to Apache Commons IO or other classes should be done later. Otherwise, IMHO the PR blows up only because of "clean code".

---

Side comment: `Injector.registerExistingAndInject(this);` is a neat thing. I wonder why we don't use that more often :p.

Reply:
Lets go for Globals and then refactor it afterwards, so this PR keeps its focus,